File: Client/src/app/characters/models/Character.ts
import { Skill } from './Skill';
import { Weapon } from './Weapon';
export interface Character {
  id: number;
  name: string;
  hitPoints: number;
  strength: number;
  defence: number;
  intelligence: number;
  user: { username: string };
  weapon: Weapon;
  skills: [Skill];
  type: string;
}

File: Server/dotnet-rpg/dotnet-rpg/Entities/Character.cs
using dotnet_rpg.Enums;
using System.Collections.Generic;

namespace dotnet_rpg.Entities
{
    public class Character
    {
        public int Id { get; set; }
        public string Name { get; set; } = "Character";
        public int HitPoints { get; set; } = 100;
        public int Strength { get; set; } = 10;
        public int Defence { get; set; } = 10;
        public int Intelligence { get; set; } = 10;
        public RpgClass Type { get; set; } = RpgClass.Knight;
        public User User { get; set; }
        public Weapon Weapon { get; set; }
        public List<Skill> Skills { get; set; }
    }
}